lib/ansible/modules/network/fortios/fortios_config.py

```diff
@@ -0,0 +1,229 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = """
+---
+module: fortios_address
+version_added: "2.3"
+author: "Benjamin Jolivot (@bjolivot)"
+short_description: Manage fortios firewall config
+description:
+  - This module provide management of FortiOS Devices configuration.
+options:
+  src:
+    description:
+      - target configuration file path.
+  dest:
+    description:
+      - Backup configuration file path.
+  filter:
+    description:
+      - in case of backup, if you want partial config, you can restrict by giving expected configuration path (ex. firewall address).
+    default: ""
+  state:
+    description:
+      - Specify if address need to be added or deleted.
+    required: true
+    choices: ['updated', 'backuped']
+
+notes:
+  - This module requires pyFG python library
+"""
+
+EXAMPLES = """
+- name: Backup current config
+  fortios_config:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    state: backuped
+    dest: ./backup.conf
+
+- name: Backup only address objects, log all commands to file
+  fortios_config:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    state: backuped
+    dest: ./backup_address_objects.conf
+    filter: "firewall address"
+    logfile: fortios.log
+
+- name: Update configuration from file
+  fortios_config:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    state: updated
+    src: new_configuration.conf
+
+"""
+
+RETURN = """
+running_config:
+  description: full config string
+  returned: always
+  type: string
+change_string:
+  description: The commands really executed by the module
+  returned: only if config changed
+  type: string
+"""
+import logging
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+
+#check for pyFG lib
+try:
+    from pyFG import FortiOS, FortiConfig
+    from pyFG.fortios import logger
+    from pyFG.exceptions import CommandExecutionException, FailedCommit, ForcedCommit
+    HAS_PYFG=True
+except:
+    HAS_PYFG=False
+
+# some blocks don't support update, so remove them
+NOT_UPDATABLE_CONFIG_OBJECTS=[
+    "vpn certificate local",
+]
+
+def main():
+    argument_spec = dict(
+        host      = dict(required=True ),
+        username  = dict(required=True ),
+        password  = dict(required=True, type='str', no_log=True ),
+        timeout   = dict(type='int', default=60),
+        vdom      = dict(type='str', default=None ),
+        state     = dict(required=True, choices=['backuped', 'updated']),
```

Comment:
I would change "updated" to "present". Some other network modules. e.g. ios_config, have an boolean option backup (e.g. backup: true) to make a backup before copied locally before changing.